# HG changeset patch
# User StrikerTheHedgefox <mossj32@gmail.com>
# Date 1743761222 10800
#      Fri Apr 04 07:07:02 2025 -0300
# Node ID 1d896309cd8588893a1fee07ce38e7081143e125
# Parent  bf843d16812ce6a63b18c13722ea153427707831
# EXP-Topic strikers_stuff
In FCloneTexture::GetPixels, return the source texture's pixels instead of NULL.
Prevents decals with their pic in the "graphics" folder from crashing the game when translated or using the "colors" DECALDEF token.

diff -r bf843d16812c -r 1d896309cd85 docs/zandronum-history.txt
--- a/docs/zandronum-history.txt	Fri Apr 04 06:55:21 2025 -0300
+++ b/docs/zandronum-history.txt	Fri Apr 04 07:07:02 2025 -0300
@@ -16,6 +16,7 @@
 *-	- Send inventory that inherits from CustomInventory and Inventory on full update. Fixes issues where weapon checks for inventory fails for late-joining clients, or after a map change. [StrikerTheHedgefox]
 +	- Added A_JumpTo codepointer, which jumps to a state instantly & with prediction. Useful for jumping to an ancestor actor's state without the issues A_Jump with a 256 chance has. [StrikerTheHedgefox]
 +	- Backported A_WeaponOffset from ZDoom Revision 115dbd0, with alterations necessary for Zandronum's current codebase. [StrikerTheHedgefox]
+-	- Fix a crash that can occur when using the "colors" DECALDEF token, when the decal's image was in the "graphics" folder. [StrikerTheHedgefox]
 
 3.2
 ---
diff -r bf843d16812c -r 1d896309cd85 src/gl/textures/gl_texture.cpp
--- a/src/gl/textures/gl_texture.cpp	Fri Apr 04 06:55:21 2025 -0300
+++ b/src/gl/textures/gl_texture.cpp	Fri Apr 04 07:07:02 2025 -0300
@@ -674,7 +674,7 @@
 
 const BYTE *FCloneTexture::GetPixels ()
 {
-	return NULL;
+	return SourcePic->GetPixels(); // [JM] Return the original texture's pixels instead of returning NULL. Prevents a particularly nasty crash in decal code.
 }
 
 void FCloneTexture::Unload ()
